tests/ui/unit/filtersSpec.js

```diff
@@ -1,19 +1,244 @@
-
-//describe('filter', function() {
-//  beforeEach(module('myApp.filters'));
-//
-//
-//  describe('interpolate', function() {
-//    beforeEach(module(function($provide) {
-//      $provide.value('version', 'TEST_VER');
-//    }));
-//
-//
-//    it('should replace VERSION', inject(function(interpolateFilter) {
-//      expect(interpolateFilter('before %VERSION% after')).toEqual('before TEST_VER after');
-//    }));
-//  });
-//});
+describe('linkifyURLs filter', function() {
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('linkifies a URL', function() {
+        var linkifyURLs = $filter('linkifyURLs');
+        expect(linkifyURLs('https://www.mozilla.org'))
+          .toEqual('<a href="https://www.mozilla.org" target="_blank">https://www.mozilla.org</a>');
+    });
+
+    it('does not linkify a non-URL', function() {
+        var linkifyURLs = $filter('linkifyURLs');
+        expect(linkifyURLs('h tee tee pee ess')).toEqual('h tee tee pee ess');
+    });
+
+    it('linkifies a mix of URL and non-URL', function() {
+        var linkifyURLs = $filter('linkifyURLs');
+        expect(linkifyURLs('This is a test: https://www.mozilla.org Did I pass?'))
+          .toEqual('This is a test: <a href="https://www.mozilla.org" target="_blank">https://www.mozilla.org</a> Did I pass?');
+    });
+});
+
+describe('linkifyBugs filter', function() {
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('linkifies a Bug', function() {
+        var linkifyBugs = $filter('linkifyBugs');
+        expect(linkifyBugs('Bug 123456'))
+          .toEqual('Bug <a href="https://bugzilla.mozilla.org/show_bug.cgi?id=123456" data-bugid=123456 title=bugzilla.mozilla.org>123456</a>');
+    });
+
+    it('linkifies a PR', function() {
+        var linkifyBugs = $filter('linkifyBugs');
+        expect(linkifyBugs('PR#123456'))
+          .toEqual('PR#<a href="https://github.com/mozilla-b2g/gaia/pull/123456" data-prid=123456 title=github.com>123456</a>');
+    });
+});
+
+describe('initials filter', function() {
+    var $filter;
+    beforeEach(module('treeherder'));
+    beforeEach(inject(function(_$filter_) {
+        $filter = _$filter_;
+    }));
+
+    it('initializes a one-word name', function() {
+        var initials = $filter('initials');
+        expect(initials('Starscream')).toEqual('<span class="label label-initials">S</span>');
+    });
+
+    it('initializes a two-word name', function() {
+        var initials = $filter('initials');
+        expect(initials('Optimus Prime')).toEqual('<span class="label label-initials">OP</span>');
+    });
+
+    it('initializes a three-word name', function() {
+        var initials = $filter('initials');
+        expect(initials('Some Other Transformer')).toEqual('<span class="label label-initials">ST</span>');
```

Comment:
<peter-griffin-voice>Heh heh heh...  robots in disguise...</peter-griffin-voice>